File: Phase#1/Week#7/b.cpp
#include <iostream>
#include <algorithm>
#include <vector>

#define endl "\n"
#define ll long long

using namespace std;

void fastIO() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
}

int main() {
    fastIO();

    int n, m;
    cin >> n >> m;

    vector<int> arr1(n);
    vector<int> arr2(m);

    for (int i = 0; i < n; i++) {
        cin >> arr1[i];
    }
    for (int i = 0; i < m; i++) {
        cin >> arr2[i];
    }

    sort(arr1.begin(), arr1.end());
    sort(arr2.begin(), arr2.end());

    if (n > m) {
        swap(arr1, arr2);
        swap(n, m);
    }

    int minn = 1e9;

    for (int i : arr1) {
        auto it = lower_bound(arr2.begin(), arr2.end(), i);
        if (it != arr2.end()) {
            minn = min(minn, abs(i - *it));
        }
        if (it != arr2.begin()) {
            it--;
            minn = min(minn, abs(i - *it));
        }
    }

    cout << minn << endl;
}

File: Phase#1/Week#7/b.py
#!/usr/bin/python3

from bisect import bisect_left
if __name__ == '__main__':
    t = 1
    # t = int(input())

    for _ in range(t):
        n, m = map(int, input().split())
        arr1 = sorted(map(int, input().split()))
        arr2 = sorted(map(int, input().split()))

        if n > m:
            n, m = m, n
            arr1, arr2 = arr2, arr1

        minn = float('inf')
        for i in arr1:
            idx = bisect_left(arr2, i)
            if idx < m:
                minn = min(minn, abs(arr2[idx] - i))
            idx -= 1
            if idx >= 0:
                minn = min(minn, abs(arr2[idx] - i))

        print(minn)
